frontend/public/module/k8s/cluster-settings.ts

```diff
@@ -39,11 +39,21 @@ export const getSortedUpdates = (cv: ClusterVersionKind): ClusterUpdate[] => {
   }
 };
 
-export const getAvailableClusterChannels = () => ({
-  'stable-4.6': 'stable-4.6',
-  'fast-4.6': 'fast-4.6',
-  'candidate-4.6': 'candidate-4.6',
-});
+export const getAvailableClusterChannels = (cv) => {
+  if (cv?.status?.desired?.channels) {
+    return cv?.status?.desired?.channels.reduce((o, val) => {
```

Comment:
No need for optional chaining here since you check above in the `if` condition

```suggestion
    return cv.status.desired.channels.reduce((o, val) => {
```

Reply:
Fixed.  Thanks, @spadgett.